ark-plonk-book/src/chapter_1.md

```diff
@@ -0,0 +1,373 @@
+# Chapter 1: Background Material
+
+## Groups
+A binary operation $*$ on a set $G$ is a mapping from $G\times G$ to $G$, which associates to elements $x$ and $y$ of $G$ a third element $x*y$ of $G$. 
+
+#### Definition: 
+A group $(G, *)$ consists of a set $G$ together with a binary operation $*$ for which the following properties are satisfied: 
+
+* Associativity: $(x*y)*z=x*(y*z),\; \forall x,y,z\in G $ 
+* Neutral element: $\exists! \; e \in G,\; e*x=x=x*e,\; \forall x\in G$
+* Inverse element: $\forall x\in G, \; \exists! \; x'\in G,\; x*x'=e=x'*x$ where $e$ is the neutral element of $G$. 
+
+A group $G$ is Abelian (or commutative) if: $x*y=y*x,\forall x,y\in G$
+
+### Cyclic groups:
+ 
+#### Definition: 
+A group $G$ is said to be cyclic, with generator $x$, if every element of $G$ is of
+the form $x^n$ for some integer $n$.
+
+## Fields
+ 
+A field $F$ is a set with two binary operations $+$ and $*$ that satisfies the following field axioms:
+ 
+* Closure under addition:  $\forall x,y\in F,\; x+y\in F$
+* Closure under multiplication:  $\forall x,y\in F,\; x*y\in F$
+* $(F, +)$ is a commutative and associative group with identity element $0$
+* $(F-\{0\},* )$ is a commutative and associative group with identity element $1$
+* Additive inverses:  $\forall x\in F,  y\in F$ such that  $x+y=y+x=0$
+* Multiplication inverses:  $\forall x\in F$ such that $x\neq 0, \; \exists \; y\in F$ such that $x*y=y*x=1$, 
+$y$ is called the multiplicative inverse of $x$ and is denoted $x^{-1}$ or $\dfrac{1}{x}$
+* The distributive law:  $\forall x,y,z\in F,\; x*(y+z)=x*y+x*z$
+
+
+### Finite Fields:
+
+
+A finite field is a field $F_q$ with a finite number of elements. The order of a finite field $$|F_q|=q=p^k$$ 
+for some integer $k\geq 1$ and $p$ prime equals the number of elements in the field.
+
+## Elliptic curves
+An Elliptic curve $E$ is a mathematical object defined over a field $F_q$ and generally expressed in the following Weierstrass form:
+$$y^2=x^3+ax+b$$ 
+for some $a,b\in F_q$ where $(x,y)$ are called "affine coordinates" 
```

Comment:
Maybe we can say something like "an elliptic curve is the set of solutions to a cubic equation over `F_q` of the given kind:", something that indicates that this equation is constraining `F_q \times F_q`.

Reply:
Also, if you mean to give a more general definition here, then probably better to have `F` as a general field instead of `F_q`.

Reply:
Good idea to use `F` here and once we restrict the topic to finite fields start using `F_q`

Reply:
Yes, will probably replace it by ```F```and use the term ```F_q```only when I talk about EC over finite fields. I will add the part about solutions of the equation in the curve, thank you for suggesting